python/lib/sift_py/report_templates/service.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import List, Optional, cast
+
+from google.protobuf.field_mask_pb2 import FieldMask
+from sift.report_templates.v1.report_templates_pb2 import (
+    CreateReportTemplateRequest,
+    CreateReportTemplateRequestClientKeys,
+    GetReportTemplateRequest,
+    GetReportTemplateResponse,
+    ReportTemplate,
+    ReportTemplateRule,
+    ReportTemplateTag,
+    UpdateReportTemplateRequest,
+)
+from sift.report_templates.v1.report_templates_pb2_grpc import ReportTemplateServiceStub
+
+from sift_py._internal.time import to_timestamp_nanos, to_timestamp_pb
+from sift_py.grpc.transport import SiftChannel
+from sift_py.ingestion.config.yaml.load import load_report_templates
+from sift_py.report_templates.config import ReportTemplateConfig
+
+
+class ReportTemplateService:
+    _report_template_service_stub: ReportTemplateServiceStub
+
+    def __init__(self, channel: SiftChannel):
+        self._report_template_service_stub = ReportTemplateServiceStub(channel)
+
+    def create_or_update_report_template(self, config: ReportTemplateConfig):
+        if not config.template_client_key:
+            raise Exception(f"Report template {config.name} requires a template_client_key")
+
+        report_template = self._get_report_template_by_client_key(config.template_client_key)
+        if report_template:
+            self._update_report_template(config, report_template)
+        else:
+            self._create_report_template(config)
+
+    def get_report_template(
+        self, client_key: str = "", id: str = ""
+    ) -> Optional[ReportTemplateConfig]:
+        report_template = None
+        if not client_key and not id:
+            raise ValueError("Either client_key or id must be provided")
+
+        if id:
+            report_template = self._get_report_template_by_id(id)
+        elif client_key:
+            report_template = self._get_report_template_by_client_key(client_key)
+
+        if not report_template:
+            raise Exception(f"Report template with client key {client_key} or id {id} not found.")
+
+        return ReportTemplateConfig(
+            name=report_template.name,
+            template_client_key=report_template.client_key,
+            organization_id=report_template.organization_id,
+            tags=[tag.tag_name for tag in report_template.tags],
+            description=report_template.description,
+            rule_client_keys=[rule.client_key for rule in report_template.rules],
+            archived_date=to_timestamp_nanos(report_template.archived_date).to_pydatetime(),
```

Comment:
`to_pydatetime` looks like its on its way to be deprecated based on docs. We can easily convert a pb timestamp to a python datetime using the [ToDatetime](https://googleapis.dev/python/protobuf/latest/google/protobuf/timestamp_pb2.html#google.protobuf.timestamp_pb2.Timestamp.ToDatetime) method.